Add a quick recipe to create an expression mapping materials to values. (#18037)
visit-dav/visit: src/doc/python_scripting/quickrecipes.rst

Available color table names can be found via ``ColorTableNames()``.

Creating an expression that maps materials to values
----------------------------------------------------

> A use case that has come up with some VisIt users is the ability to associated a scalar with material.

Reviewer: ```suggestion
A use case that has come up with some VisIt users is the ability to associate scalar values with material numbers.
```

> The following example defines a function that creates an expression that maps material numbers to values.

Reviewer: ```suggestion
The following example defines a function that creates an expression that maps material numbers to scalar values.
```

> It take a list of tuples of material number and value.

Reviewer: ```suggestion
It takes a list of *pairs* of material number and scalar value.
```

> The material number of the last tuple is ignored and the value is used for any unspecified materials.

Reviewer: ```suggestion
The material number of the last pair is ignored.
Its value is used for any unspecified materials.
```

The expresion generated by calling the function is then used in a Psuedocolor plot.:

.. literalinclude: ../../test/tests/quickrecipes/expressions.py
    :language: Python
    :start-after: # mapMaterialsToValues {
    :end-before: # mapMaterialsToValues }

> .. figure::  https://media.githubusercontent.com/media/visit-dav/visit/develop/test/baseline/quickrecipes/expressions/materials_to_values.png

Reviewer: I am not sure the *figure* directive supports URL image regs. I know the *image* directive does, `.. image::` I guess we can commit and see.

And, since the ref'd image won't be on `develop` until the PR is merged, it is indeed hard to tell 😉
